automation/test.sh

```diff
@@ -297,7 +297,7 @@ set -e
 echo "Nodes are ready:"
 kubectl get nodes
 
-make cluster-sync
+ionice --class idle make cluster-sync
```

Comment:
reading a little about this (total noob atm) it seems some people set the etcd to higher priority instead:
https://etcd.io/docs/v3.2/tuning/#disk

Reply:
As etcd is running inside the VMs - the worker nodes would not be aware of the process in order to prioritize it.

Reply:
TBH I like this solution more, I'm sure we could find a way to achieve this in kvci
https://github.com/search?q=repo%3Aopenshift%2Fcluster-etcd-operator%20ionice&type=code

Reply:
Yes but if the CI node disk is saturated from other jobs - the priority in the kubevirtci VM won't help. 

Giving priority to etcd within the kubevirtci VMs will probably help with getting priority over the test workloads which is an improvement we should look at as well.

Reply:
I see.. normally I'd expect that the disk is sliced for the different jobs in such a way that this interference would not be a huge problem. So 
> CI node disk is saturated from other jobs

really surprises me. Do you know how the storage is set up?

Maybe it's best that we ask for an extra volume/carve one from what we're given just for etcd.
This is the recommendation that comes up in etcd tuning guides:
https://www.suse.com/support/kb/doc/?id=000020100
https://docs.openshift.com/container-platform/4.16/scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.html

Reply:
> I see.. normally I'd expect that the disk is sliced for the different jobs in such a way that this interference would not be a huge problem. So
> 
> > CI node disk is saturated from other jobs
> 
> really surprises me. Do you know how the storage is set up?
> 
The CI nodes have a single 1TB disk each that is used by the OS and test pods.

> Maybe it's best that we ask for an extra volume/carve one from what we're given just for etcd.
> This is the recommendation that comes up in etcd tuning guides:

Again etcd tuning inside of the kubevirtci nodes will not solve for IO competition on the bare-metal worker nodes but it is worth looking at in parallel with this change. etcd IO is not causing a problem for the bare-metal nodes as most of the lanes are running etcd in memory. 

This change is to avoid saturating the disk on the bare-metal nodes themselves.

Reply:
How is that 1TB disk sliced for jobs? some kind of hostpath provisioner? This could be the source of the problem

>Again etcd tuning inside of the kubevirtci nodes will not solve for IO competition on the bare-metal worker

I am saying propagate a separate volume to the etcd all the way up from the prow job
(possibly a volume thats not related to the 1TB attached to the CI node)

Reply:
>I am saying propagate a separate volume to the etcd all the way up from the prow job
(possibly a volume thats not related to the 1TB attached to the CI node)

This sounds good but please keep in mind we see problems now and we have many people complaining about the current CI. So whatever we can do to help in short term should be done. Long term solution can be implemented while we have the short term applied.

Reply:
That sounds good and I have no issue to merge this first, but it's even better that we started discussing the CI storage/etcd setup here

BTW - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12511/pull-kubevirt-e2e-k8s-1.30-sig-storage/1820768392882163712

Reply:
> How is that 1TB disk sliced for jobs? some kind of hostpath provisioner? This could be the source of the problem

They are just basically emptyDirs. 

> 
> > Again etcd tuning inside of the kubevirtci nodes will not solve for IO competition on the bare-metal worker
> 
> I am saying propagate a separate volume to the etcd all the way up from the prow job (possibly a volume thats not related to the 1TB attached to the CI node)

We don't have other disks or storage to use.  Setting etcd with a higher priority in kubevirtci is something we can try though.

Reply:
> They are just basically emptyDirs.

:open_mouth: I think this is what makes is possible for jobs to impact each other. If we had some kind of cloud storage like https://aws.amazon.com/ebs/general-purpose/ I think we would be okay.
Is there maybe an equivalent of this in the cloud we're using?

BTW cc @awels @mhenriks, for additional thoughts on the CI setup

Reply:
>That sounds good and I have no issue to merge this first, but it's even better that we started discussing the CI storage/etcd setup here

>BTW - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12511/pull-kubevirt-e2e-k8s-1.30-sig-storage/1820768392882163712

Yes, there is also option of in-memory etcd. Can we open an issue and put options there?


Btw, we could do some logical volumes but otherwise we don't have much storage options afaik

Reply:
> > That sounds good and I have no issue to merge this first, but it's even better that we started discussing the CI storage/etcd setup here
> 
> > BTW - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12511/pull-kubevirt-e2e-k8s-1.30-sig-storage/1820768392882163712
> 
> Yes, there is also option of in-memory etcd. Can we open an issue and put options there?
> 
> Btw, we could do some logical volumes but otherwise we don't have much storage options afaik

Yes please remember that CI is a limited resource and that we are basically at our limit.

Reply:
That's okay, I think we still have options to pool that 1TB disk, as long as we have it exposed to the k8s cluster. That should at least be better than the emptydirs, maybe something like topolvm?